src/main/java/org/opensearch/security/identity/SecurityScheduledJobIdentityManager.java

```diff
@@ -0,0 +1,138 @@
+        this.cs = cs;
+        this.client = client;
+        this.threadPool = threadPool;
+        this.securityIndices = new SecurityIndices(client, cs);
+    }
+
+    @Override
+    public void saveUserDetails(String jobId, String indexName) {
+        if (!securityIndices.doesScheduledJobIdentityIndexExists()) {
+           securityIndices.initScheduledJobIdentityIndex(ActionListener.wrap(response -> {
+                if (response.isAcknowledged()) {
+                    logger.info("Created {} with mappings.", SCHEDULED_JOB_IDENTITY_INDEX);
+                    createScheduledJobIdentityEntry(jobId, indexName);
+                } else {
+                    logger.warn("Created {} with mappings call not acknowledged.", SCHEDULED_JOB_IDENTITY_INDEX);
+                    throw new OpenSearchSecurityException(
+                        "Created " + SCHEDULED_JOB_IDENTITY_INDEX + " with mappings call not acknowledged."
+                    );
+                }
+           }, exception -> new OpenSearchSecurityException(
+               "Created " + SCHEDULED_JOB_IDENTITY_INDEX + " with mappings call failed."
+           )));
+        } else {
+            createScheduledJobIdentityEntry(jobId, indexName);
+        }
+    }
+
+    private void createScheduledJobIdentityEntry(String jobId, String indexName) {
+        SearchRequest searchRequest = new SearchRequest().indices(SCHEDULED_JOB_IDENTITY_INDEX);
+        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery()
+                .must(QueryBuilders.matchQuery("job_id", jobId))
+                .must(QueryBuilders.matchQuery("job_index", indexName));
+        searchRequest.source(SearchSourceBuilder.searchSource().query(boolQuery));
+
+        client.search(
+            searchRequest,
+            ActionListener
+                .wrap(
+                    response -> indexScheduledJobIdentity(response, jobId, indexName),
+                    exception -> new OpenSearchSecurityException(
+                        "Exception received while querying for " + jobId + " in " + SCHEDULED_JOB_IDENTITY_INDEX
+                    )
+                )
+        );
+    }
+
+    private void indexScheduledJobIdentity(
+            SearchResponse response,
+            String jobId,
+            String indexName
+    ) throws IOException {
+        long totalHits = response.getHits().getTotalHits().value;
+        if (totalHits > 1) {
+            // Should not happen
+            logger.warn("Multiple scheduled job identities already exists in " + SCHEDULED_JOB_IDENTITY_INDEX + " for job with jobId " + jobId);
+        } else if (totalHits == 1) {
+            logger.info("Scheduled Job Identity already exists in " + SCHEDULED_JOB_IDENTITY_INDEX + " for job with jobId " + jobId);
+        } else {
+            final User user = threadPool.getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
```

Comment:
@davidlago Made a good comment that if an extension/plugin is writing to its Jobs Index that the user in the threadcontext may not be the right user to persist for the job identity index. I'm looking into ways that the correct user information can be provided.